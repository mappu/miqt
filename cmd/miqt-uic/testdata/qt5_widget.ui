<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <widget class="QLabel" name="label">
   <property name="geometry">
    <rect>
     <x>20</x>
     <y>20</y>
     <width>231</width>
     <height>21</height>
    </rect>
   </property>
   <property name="text">
    <string>QWidget as base class</string>
   </property>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
