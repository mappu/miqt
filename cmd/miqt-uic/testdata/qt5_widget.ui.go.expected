// Generated by miqt-uic. To update this file, edit the .ui file in
// Qt Designer, and then run 'go generate'.
//
//go:generate miqt-uic go_generate_args

package test_package_name

import (
	"github.com/mappu/miqt/qt"
)

type FormUi struct {
	Form  *qt.QWidget
	label *qt.QLabel
}

// NewFormUi creates all Qt widget classes for Form.
func NewFormUi() *FormUi {
	ui := &FormUi{}
	ui.Form = qt.NewQWidget(nil)
	ui.Form.SetObjectName("Form")
	ui.Form.Resize(400, 300)
	ui.label = qt.NewQLabel(ui.Form)
	ui.label.SetObjectName("label")
	ui.label.SetGeometry(qt.NewQRect(20, 20, 231, 21))

	ui.Retranslate()

	return ui
}

// Retranslate reapplies all text translations.
func (ui *FormUi) Retranslate() {
	ui.Form.SetWindowTitle(qt.QCoreApplication_Tr("Form"))
	ui.label.SetText(qt.QWidget_Tr("QWidget as base class"))
}
